xbmc/video/VideoThumbLoader.cpp

```diff
@@ -223,68 +223,92 @@ static void SetupRarOptions(CFileItem& item, const std::string& path)
-  if (type == MediaTypeEpisode)
-  {
-    ret = {"thumb"};
-    for (auto& artType : advancedSettings->m_videoEpisodeExtraArt)
-    {
-      if (find(ret.begin(), ret.end(), artType) == ret.end())
-        ret.push_back(artType);
-    }
-  }
-  else if (type == MediaTypeTvShow)
-  {
-    ret = {"poster", "fanart", "banner"};
-    for (auto& artType : advancedSettings->m_videoTvShowExtraArt)
-    {
-      if (find(ret.begin(), ret.end(), artType) == ret.end())
-        ret.push_back(artType);
-    }
-  }
-  else if (type == MediaTypeSeason)
-  {
-    ret = {"poster", "fanart", "banner"};
-    for (auto& artType : advancedSettings->m_videoTvSeasonExtraArt)
-    {
-      if (find(ret.begin(), ret.end(), artType) == ret.end())
-        ret.push_back(artType);
-    }
-  }
-  else if (type == MediaTypeMovie)
+  int artworkLevel = CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
+    CSettings::SETTING_VIDEOLIBRARY_ARTWORK_LEVEL);
+  if (artworkLevel == CSettings::VIDEOLIBRARY_ARTWORK_LEVEL_NONE)
   {
-    ret = {"poster", "fanart"};
-    for (auto& artType : advancedSettings->m_videoMovieExtraArt)
-    {
-      if (find(ret.begin(), ret.end(), artType) == ret.end())
-        ret.push_back(artType);
-    }
+    return {};
   }
-  else if (type == MediaTypeVideoCollection)
+
+  std::vector<std::string> result = GetArtTypeDefault(type);
+  if (artworkLevel != CSettings::VIDEOLIBRARY_ARTWORK_LEVEL_CUSTOM)
   {
-    ret = {"poster", "fanart"};
-    for (auto& artType : advancedSettings->m_videoMovieSetExtraArt)
-    {
-      if (find(ret.begin(), ret.end(), artType) == ret.end())
-        ret.push_back(artType);
-    }
+    return result;
   }
-  else if (type == MediaTypeMusicVideo)
+
+  auto settings = artTypeSettings.find(type);
+  if (settings == artTypeSettings.end())
+    return result;
+
+  for (auto& artType : GetSettingListAsString(settings->second))
   {
-    ret = {"poster", "fanart"};
-    for (auto& artType : advancedSettings->m_videoMusicVideoExtraArt)
-    {
-      if (find(ret.begin(), ret.end(), artType) == ret.end())
-        ret.push_back(artType);
-    }
+    if (find(result.begin(), result.end(), artType) == result.end())
+      result.push_back(artType);
   }
-  else if (type.empty()) // unknown, just the basics
-    ret = { "poster", "fanart", "banner", "thumb" };
 
-  return ret;
+  return result;
+}
+
+bool CVideoThumbLoader::IsValidArtType(const std::string& potentialArtType)
+{
+  return !potentialArtType.empty() && potentialArtType.length() <= 25 &&
+    std::find_if_not(
+      potentialArtType.begin(), potentialArtType.end(),
+      StringUtils::isasciialphanum
+    ) == potentialArtType.end();
+}
+
+bool CVideoThumbLoader::IsArtTypeInWhitelist(const std::string& artType, const std::vector<std::string>& whitelist)
+{
+  // whitelist contains art "families", 'fanart' also matches 'fanart1', 'fanart2', and so on
+  std::string compareArtType = artType;
+  StringUtils::TrimRight(compareArtType, "0123456789");
```

Comment:
```
 if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
          CSettings::SETTING_VIDEOLIBRARY_ARTWORK_LEVEL) ==
      CSettings::VIDEOLIBRARY_ARTWORK_LEVEL_CUSTOM)
    StringUtils::TrimRight(compareArtType, "0123456789");
```
Will avoid the pickup of families of art types e.g. fanart1, fanart2 etc. when artlevel is basic